src/Microsoft.Data.SqlClient/netcore/src/Common/src/Microsoft/Data/Common/AdapterUtil.cs

```diff
@@ -379,24 +379,33 @@ internal static Exception StreamClosed([CallerMemberName] string method = "")
 
         internal static string BuildQuotedString(string quotePrefix, string quoteSuffix, string unQuotedString)
         {
-            var resultString = new StringBuilder();
+            var resultString = new StringBuilder(unQuotedString.Length + quoteSuffix.Length + quoteSuffix.Length);
+            AppendQuotedString(resultString, quotePrefix, quoteSuffix, unQuotedString);
+            return resultString.ToString();
```

Comment:
The constructor argument changes the initial capacity so that a single allocation of at least that length is used for the first segment. If you know how much space you need it's a good idea to use this ctor. Even if you can't presize it's often a good idea to have a guess because the less segments you add the less heap fragmentation you cause. Additional allocations will use the internal sizing rules if they are added so defaults aren't changed.

There are no cons to returning a stringbuilder and having the caller deal with calling ToString on it. I chose to maintain the call pattern and return a string.